### DISCUSSION TITLE
Nginx and certbot SSL

### DISCUSSION THREAD

**Discussion by smsorin** [nginx-proxy/nginx-proxy#2367] [Q&A]
Hi, I'm a bit new at this. I've followed the guides and I have a nginx-proxy and a portainer instance. The server is private, so I have a script with docker to get certificates with the DNS challenge(the default stuff from [the site](https://eff-certbot.readthedocs.io/en/stable/install.html#alternative-1-docker) ). It creates this directory with the format `domain.tld/privkey.pem` and all the other files. Now nginx-proxy needs the certificates as domain.tld.cer and .key. I could copy/move stuff around, but that will break when the certificates are renewed.

I can see in /etc/nginx/conf.d/default.conf an entry for portainer but it says
```
   # No certificate found for this vhost, so force nginx to emit a TLS error if
   # the client connects via https.
```
So clearly, it can't find the certificates I've provided.

Did I miss a step?

 Is there a script/flag/option that allows nginx-proxy to find certificates in this format?

**Comment by buchdag** [comment]
Hi @smsorin 

Unfortunately no there is no option to point nginx-proxy to a fully custom certificate path and name for each virtual host. The closest we have at the moment is the `CERT_NAME` environment variable, but that will only allow you to point nginx-proxy to a specific file name, not a different extension or path (when used on  nginx proxy will look for `/etc/nginx/certs/${CERT_NAME}.crt` and `/etc/nginx/certs/${CERT_NAME}.key`).

[certbot has hooks on renewal](https://eff-certbot.readthedocs.io/en/stable/using.html#renewing-certificates) that should allow you to script the necessary actions (moving the renewed certificates to their correct path and name, reloading nginx).
